src/runtime/graph/graph_runtime_factory.h

```diff
@@ -0,0 +1,133 @@
+
+#include <tvm/runtime/c_runtime_api.h>
+#include <tvm/runtime/module.h>
+#include <tvm/runtime/ndarray.h>
+#include <tvm/runtime/packed_func.h>
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace tvm {
+namespace runtime {
+
+class TVM_DLL GraphRuntimeFactory : public runtime::ModuleNode {
+ public:
+  /*!
+   * \brief Initialize the GraphRuntimeFactory with graph and context.
+   * \param graph_json The execution graph.
+   * \param params The params of graph.
+   * \param kind The runtime kind to be created.
+   */
+  void Init(const std::string& kind, const std::string& graph_json,
+            const std::unordered_map<std::string, tvm::runtime::NDArray>& params,
+            const std::string& module_name = "default");
+
+  /*!
+   * \brief Import other GraphRuntimeFactory module.
+   * \param other The GraphRuntimeFactory module we want to import.
+   */
+  void ImportModule(Module other);
+
+  /*!
+   * \brief Get member function to front-end
+   * \param name The name of the function.
+   * \param sptr_to_self The pointer to the module node.
+   * \return The corresponding member function.
+   */
+  virtual PackedFunc GetFunction(const std::string& name, const ObjectPtr<Object>& sptr_to_self);
+
+  /*!
+   * \return The type key of the executor.
+   */
+  const char* type_key() const override { return "GraphRuntimeFactory"; }
+
+  /*!
+   * \brief Save the module to binary stream.
+   * \param stream The binary stream to save to.
+   */
+  void SaveToBinary(dmlc::Stream* stream) override;
+
+  /*!
+   * \brief Create a specific runtime module
+   * \param module The module we will be used for creating runtime
+   * \param ctxs The context of the host and devices where graph nodes will be
+   *  executed on.
+   * \return created runtime module
+   */
+  Module RuntimeCreate(Module module, const std::vector<TVMContext>& ctxs);
+
+  /*!
+   * \brief Select the specific module
+   * \param name The name of the module
+   * \return selected module
+   */
+  Module SelectModule(const std::string& name);
+
+  const std::string& GetJson() const { return graph_json_; }
+
+  std::unordered_map<std::string, tvm::runtime::NDArray> GetParams() const { return params_; }
+
+  Module GetLib() const {
+    CHECK_GT(this->imports().size(), 0);
+    return this->imports_[0];
+  }
+
+  const std::string& GetKind() const { return kind_; }
+
+  const std::string& GetModuleName() const { return module_name_; }
+
+  const std::vector<std::string>& GetGraphRuntimeFactoryModuleList() const {
+    return graph_runtime_factory_module_list_;
+  }
+
+  void SetGraphRuntimeFactoryModuleList(
+      const std::vector<std::string>& graph_runtime_factory_module_list) {
+    graph_runtime_factory_module_list_ = graph_runtime_factory_module_list;
+  }
+
+ protected:
+  /*! \brief The execution graph. */
+  std::string graph_json_;
+  /*! \brief The params. */
+  std::unordered_map<std::string, tvm::runtime::NDArray> params_;
```

Comment:
I have one small query here. Do these params contains for all the modules imported or only for the first one ?

Reply:
The logic is `GraphRuntimeFacotryModule` will contain one `graph runtime module`. Then it will contain other `GraphRuntimeFacotryModule` (which could contain another one `graph runtime module`). So, let us take `test_multi_model` for example, `resnet18 graph runtime factory module` contains `resnet18 graph runtime module`, `resnet18 params` (we talked here), `resnet18 json`. Then we import another `resnet50 graph runtime factory module`. Inside it, which contains `resnet18 graph runtime module`, `resnet18 params`, `resnet18 json`. Does it make sense to you now?

Reply:
Thanks for explaining in detail! 
So to confirm once again, this params_ will contain only the first module imported at any point of time, right?
In your example case , it will be always `resnet18 params`, correct?

Reply:
Every `graph runtime factory module` will has its own unique `library` / `param` / `json`. So, for resnet18 graph runtime factory module, it has `resnet18 params`. for resnet50 graph runtime factory module, it has `resnet50 params`.

Reply:
Thanks a lot! I got it now.